sentry_sdk/integrations/anthropic.py

```diff
@@ -75,105 +75,150 @@ def _calculate_token_usage(result, span):
     record_token_usage(span, input_tokens, output_tokens, total_tokens)
 
 
+def _sentry_patched_create_common(f, *args, **kwargs):
+    # type: (Any, *Any, **Any) -> Any
+    if "messages" not in kwargs:
+        return f(*args, **kwargs)
+
+    try:
+        iter(kwargs["messages"])
+    except TypeError:
+        return f(*args, **kwargs)
+
+    messages = list(kwargs["messages"])
+    model = kwargs.get("model")
+
+    span = sentry_sdk.start_span(
+        op=OP.ANTHROPIC_MESSAGES_CREATE,
+        description="Anthropic messages create",
+        origin=AnthropicIntegration.origin,
+    )
+    span.__enter__()
+
+    try:
+        result = yield f, args, kwargs
+    except Exception as exc:
+        _capture_exception(exc)
+        span.__exit__(None, None, None)
+        raise exc from None
+
+    integration = sentry_sdk.get_client().get_integration(AnthropicIntegration)
+
+    with capture_internal_exceptions():
+        span.set_data(SPANDATA.AI_MODEL_ID, model)
+        span.set_data(SPANDATA.AI_STREAMING, False)
+        if should_send_default_pii() and integration.include_prompts:
+            span.set_data(SPANDATA.AI_INPUT_MESSAGES, messages)
+        if hasattr(result, "content"):
+            if should_send_default_pii() and integration.include_prompts:
+                span.set_data(
+                    SPANDATA.AI_RESPONSES,
+                    list(
+                        map(
+                            lambda message: {
+                                "type": message.type,
+                                "text": message.text,
+                            },
+                            result.content,
+                        )
+                    ),
+                )
+            _calculate_token_usage(result, span)
+            span.__exit__(None, None, None)
+        elif hasattr(result, "_iterator"):
+            old_iterator = result._iterator
+
+            def new_iterator():
+                # type: () -> Iterator[MessageStreamEvent]
+                input_tokens = 0
+                output_tokens = 0
+                content_blocks = []
+                with capture_internal_exceptions():
+                    for event in old_iterator:
+                        if hasattr(event, "type"):
+                            if event.type == "message_start":
+                                usage = event.message.usage
+                                input_tokens += usage.input_tokens
+                                output_tokens += usage.output_tokens
+                            elif event.type == "content_block_start":
+                                pass
+                            elif event.type == "content_block_delta":
+                                content_blocks.append(event.delta.text)
```

Comment:
You might want to change this here as it will break function calling.

https://github.com/getsentry/sentry-python/pull/3615

Reply:
will do so while merging